docs/proposed/config-split.md

```diff
@@ -0,0 +1,469 @@
+//#define MBEDTLS_SSL_DEBUG_ALL  
+#define MBEDTLS_SSL_ENCRYPT_THEN_MAC  
+#define MBEDTLS_SSL_EXTENDED_MASTER_SECRET  
+#define MBEDTLS_SSL_KEEP_PEER_CERTIFICATE  
+#define MBEDTLS_SSL_RENEGOTIATION  
+#define MBEDTLS_SSL_MAX_FRAGMENT_LENGTH  
+//#define MBEDTLS_SSL_RECORD_SIZE_LIMIT  
+#define MBEDTLS_SSL_PROTO_TLS1_2  
+#define MBEDTLS_SSL_PROTO_TLS1_3  
+#define MBEDTLS_SSL_TLS1_3_COMPATIBILITY_MODE  
+#define MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_ENABLED  
+#define MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_EPHEMERAL_ENABLED  
+#define MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_EPHEMERAL_ENABLED  
+//#define MBEDTLS_SSL_EARLY_DATA  
+#define MBEDTLS_SSL_PROTO_DTLS  
+#define MBEDTLS_SSL_ALPN  
+#define MBEDTLS_SSL_DTLS_ANTI_REPLAY  
+#define MBEDTLS_SSL_DTLS_HELLO_VERIFY  
+//#define MBEDTLS_SSL_DTLS_SRTP  
+#define MBEDTLS_SSL_DTLS_CLIENT_PORT_REUSE  
+#define MBEDTLS_SSL_SESSION_TICKETS  
+#define MBEDTLS_SSL_SERVER_NAME_INDICATION  
+//#define MBEDTLS_SSL_VARIABLE_BUFFER_LENGTH  
+//#define MBEDTLS_USE_PSA_CRYPTO  
+//#define MBEDTLS_X509_TRUSTED_CERTIFICATE_CALLBACK  
+//#define MBEDTLS_X509_REMOVE_INFO  
+#define MBEDTLS_X509_RSASSA_PSS_SUPPORT  
+
+
+* SECTION "Mbed TLS modules"  
+#define MBEDTLS_DEBUG_C  
+#define MBEDTLS_ERROR_C  
+#define MBEDTLS_NET_C  
+#define MBEDTLS_PKCS7_C  
+#define MBEDTLS_SSL_CACHE_C  
+#define MBEDTLS_SSL_COOKIE_C  
+#define MBEDTLS_SSL_TICKET_C  
+#define MBEDTLS_SSL_CLI_C  
+#define MBEDTLS_SSL_SRV_C  
+#define MBEDTLS_SSL_TLS_C  
+#define MBEDTLS_X509_USE_C  
+#define MBEDTLS_X509_CRT_PARSE_C  
+#define MBEDTLS_X509_CRL_PARSE_C  
+#define MBEDTLS_X509_CSR_PARSE_C  
+#define MBEDTLS_X509_CREATE_C  
+#define MBEDTLS_X509_CRT_WRITE_C  
+#define MBEDTLS_X509_CSR_WRITE_C  
+
+
+* SECTION "General configuration options"  
+//#define MBEDTLS_CONFIG_FILE "mbedtls/mbedtls_config.h"  
+//#define MBEDTLS_USER_CONFIG_FILE "/dev/null"  
+
+
+* SECTION "Module configuration options"  
+//#define MBEDTLS_SSL_CACHE_DEFAULT_TIMEOUT       86400  
+//#define MBEDTLS_SSL_CACHE_DEFAULT_MAX_ENTRIES      50  
+//#define MBEDTLS_SSL_IN_CONTENT_LEN              16384  
+//#define MBEDTLS_SSL_CID_IN_LEN_MAX 32  
+//#define MBEDTLS_SSL_CID_OUT_LEN_MAX 32  
+//#define MBEDTLS_SSL_CID_TLS1_3_PADDING_GRANULARITY 16  
+//#define MBEDTLS_SSL_OUT_CONTENT_LEN             16384  
+//#define MBEDTLS_SSL_DTLS_MAX_BUFFERING             32768  
+//#define MBEDTLS_PSK_MAX_LEN               32  
+//#define MBEDTLS_SSL_COOKIE_TIMEOUT        60  
+//#define MBEDTLS_SSL_CIPHERSUITES MBEDTLS_TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,MBEDTLS_TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256  
+//#define MBEDTLS_SSL_MAX_EARLY_DATA_SIZE        1024  
+//#define MBEDTLS_SSL_TLS1_3_TICKET_AGE_TOLERANCE 6000  
+//#define MBEDTLS_SSL_TLS1_3_TICKET_NONCE_LENGTH 32  
+//#define MBEDTLS_SSL_TLS1_3_DEFAULT_NEW_SESSION_TICKETS 1  
+//#define MBEDTLS_X509_MAX_INTERMEDIATE_CA   8  
+//#define MBEDTLS_X509_MAX_FILE_PATH_LEN     512  
```

Comment:
Please add a section on `config.py` as a configuration interface. Which file(s) will it edit?